tests/foreman/api/test_classparameters.py

```diff
@@ -0,0 +1,1035 @@
+        """
+
+    @run_only_on('sat')
+    @stubbed()
+    @tier3
+    def test_positive_list_parameters_by_environment_id(self):
+        """List all the parameters for specific environment by id.
+
+        @feature: Class Parameters - List
+
+        @assert: Parameters listed for specific environment.
+
+        @status: Manual
+        """
+
+    @run_only_on('sat')
+    @stubbed()
+    @tier3
+    def test_positive_override_checkbox(self):
+        """Override the Default Parameter value.
+
+        @feature: Class Parameters - Override
+
+        @steps:
+
+        1.  Set override to True.
+        2.  Set the new valid Default Value.
+
+        @assert: Parameter Value overridden with new value.
+
+        @status: Manual
+        """
+
+    @run_only_on('sat')
+    @stubbed()
+    @tier3
+    def test_negative_override_checkbox(self):
+        """Override the Default Parameter value - override Unchecked.
+
+        @feature: Class Parameters - Override
+
+        @steps:
+
+        1.  Set override to False.
+        2.  Set the new valid Default Value.
+
+        @assert: Parameter value not allowed/disabled to override.
+
+        @status: Manual
+        """
+
+    @run_only_on('sat')
+    @stubbed()
+    @tier3
+    def test_positive_puppet_default(self):
+        """On Override, Set Puppet Default Value.
+
+        @feature: Class Parameters - Puppet Default
+
+        @steps:
+
+        1.  Set override to True.
+        2.  Set 'Use Puppet Default' to True.
+
+        @assert: Puppet Default Value applied on parameter.
+
+        @status: Manual
+        """
+
+    @run_only_on('sat')
+    @stubbed()
+    @tier1
+    def test_positive_update_parameter_type(self):
+        """Positive Parameter Update for parameter types - Valid Value.
+
+        Types - string, boolean, integer, real, array, hash, yaml, json
+
+        @feature: Class Parameters - Parameter Type
+
+        @steps:
+
+        1.  Set override to True.
+        2.  Update the Key Type to any of available.
+        3.  Set a 'valid' default Value.
+
+        @assert: Parameter Updated with a new type successfully.
+
+        @status: Manual
+        """
+
+    @run_only_on('sat')
+    @stubbed()
+    @tier1
+    def test_negative_update_parameter_type(self):
+        """Negative Parameter Update for parameter types - Invalid Value.
+
+        Types - string, boolean, integer, real, array, hash, yaml, json
+
+        @feature: Class Parameters - Parameter Type
+
+        @steps:
+
+        1.  Set override to True.
+        2.  Update the Key Type.
+        3.  Attempt to set an 'Invalid' default Value.
+
+        @assert:
+
+        1.  Parameter not updated with string type for invalid value.
+        2.  Error raised for invalid default value.
+
+        @status: Manual
+        """
+
+    @run_only_on('sat')
+    @stubbed()
+    @tier3
+    def test_positive_validate_puppet_default_value(self):
+        """Validation doesn't works on puppet default value.
```

Comment:
positive - and `doesn't work`?
Even that is true, please correct `doesn't works` on `doesn't work`